src/apis/issue.ts

```diff
@@ -5,21 +5,37 @@ import { EventRecord } from "@polkadot/types/interfaces/system";
 import { Bytes, u32 } from "@polkadot/types/primitive";
 import { DOT, H256Le, IssueRequest, PolkaBTC, Vault } from "../interfaces/default";
 import { DefaultVaultsAPI, VaultsAPI } from "./vaults";
-import { pagedIterator, sendLoggedTx } from "../utils";
+import { encodeBtcAddress, pagedIterator, sendLoggedTx } from "../utils";
 import { BlockNumber } from "@polkadot/types/interfaces/runtime";
+import { Network } from "bitcoinjs-lib";
 
 export type RequestResult = { hash: Hash; vault: Vault };
 
+export interface IssueRequestExt extends Omit<IssueRequest, "btc_address"> {
+    // network encoded btc address
+    btc_address: string;
+}
+
+export function encodeIssueRequest(req: IssueRequest, network: Network): IssueRequestExt {
+    const { btc_address, ...obj } = req;
+    return Object.assign(
+        {
+            btc_address: encodeBtcAddress(btc_address, network),
+        },
+        obj
+    ) as IssueRequestExt;
+}
+
 export interface IssueAPI {
     request(amount: PolkaBTC, vaultId?: AccountId, griefingCollateral?: DOT): Promise<RequestResult>;
-    execute(issueId: H256, txId: H256Le, txBlockHeight: u32, merkleProof: Bytes, rawTx: Bytes): Promise<boolean>;
+    execute(issueId: H256, txId: H256Le, merkleProof: Bytes, rawTx: Bytes): Promise<boolean>;
     cancel(issueId: H256): Promise<void>;
     setAccount(account?: AddressOrPair): void;
     getGriefingCollateral(): Promise<DOT>;
-    list(): Promise<IssueRequest[]>;
+    list(network: Network): Promise<IssueRequestExt[]>;
```

Comment:
We already pass the network on constructing the basic API (https://github.com/interlay/polkabtc-js/blob/dev/src/polkabtc-api.ts#L52). Let's store the network as part of the API and get the network from there so we don't have to submit it as a parameter.

Reply:
Ah yes good catch!